flow/display_list_benchmarks_metal.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/display_list_benchmarks.h"
+#include "flutter/testing/test_metal_surface.h"
+
+#include "third_party/skia/include/core/SkCanvas.h"
+
+namespace flutter {
+namespace testing {
+
+class MetalCanvasProvider : public CanvasProvider {
+ public:
+  virtual ~MetalCanvasProvider() = default;
+  void InitializeSurface(const size_t width, const size_t height) override {
+    metal_context_ = std::make_unique<TestMetalContext>();
+    metal_surface_ =
+        TestMetalSurface::Create(*metal_context_, SkISize::Make(width, height));
+    metal_surface_->GetSurface()->getCanvas()->clear(SK_ColorTRANSPARENT);
+  }
+
+  sk_sp<SkSurface> GetSurface() override {
+    if (!metal_surface_) {
+      return nullptr;
+    }
+    return metal_surface_->GetSurface();
+  }
+
+  sk_sp<SkSurface> MakeOffscreenSurface(const size_t width,
+                                        const size_t height) override {
+    // TODO(wrightgeorge): should we add an API to TestMetalSurface to create an
+    // offscreen?
```

Comment:
Or maybe it isn't obsolete? Is this a suggestion for a better way to implement this method?

Reply:
Yeah, the suggestion is whether to modify the `TestMetalSurface` class in `flutter/testing`. "Offscreen" is a little bit of a misnomer here anyway because they're all offscreens, but I'm trying to distinguish between the "main" surface and other surfaces. I can delete the comment, I don't think it really matters that much.